src/Components/Server/src/ComponentsServerOptions.cs

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Components.Builder;
+using Microsoft.AspNetCore.Components.Server.Circuits;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Components.Server
+{
+    public class ComponentsServerOptions
+    {
+        private readonly Dictionary<PathString, Type> _circuitHandlers = new Dictionary<PathString, Type>();
+
+        public IReadOnlyDictionary<PathString, Type> CircuitHandlers => _circuitHandlers;
+
+        public Type DefaultCircuitHandler { get; private set; }
+
+        public void AddCircuitHandler<THandler>(PathString path) where THandler : CircuitHandler
```

Comment:
This thing is fairly obtuse, but it sort of follows the same pattern as `StartupActions`. I couldn't figure out a way to associate each hub instance with a separate configuration. 

For features like tombstoning, it's likely that users are going to need other DI services in their handlers. I wanted to see if we could force a design that required doing a DI. But the API does feel very user hostile.